sql/core/src/main/scala/org/apache/spark/sql/columnar/InMemoryColumnarTableScan.scala

```diff
@@ -314,7 +314,7 @@ private[sql] case class InMemoryColumnarTableScan(
                 columnAccessors(i).extractTo(nextRow, i)
                 i += 1
               }
-              nextRow
+              if (attributes.isEmpty) Row.empty else nextRow
```

Comment:
If `attributes.isEmpty`, we'll always return `Row.empty`, so we don't need to create `ColumnAccessor`s, right? The [code]( https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/columnar/InMemoryColumnarTableScan.scala#L283-L291) looks weird to me, why we handle `attributes.isEmpty` this way?